pkg/adaptor/cloud/ibmcloud/provider.go

```diff
@@ -60,6 +60,23 @@ func NewProvider(config *Config) (cloud.Provider, error) {
 		return nil, fmt.Errorf("either an IAM API Key or Profile ID needs to be set")
 	}
 
+	nodeName, ok := os.LookupEnv("NODE_NAME")
+	var nodeLabels map[string]string
+	if ok {
+		var err error
+		nodeLabels, err = k8sops.NodeLabels(context.TODO(), nodeName)
```

Comment:
The necessity of `context.TODO()` here probably indicates that the `cloud.Provider` interface is not properly  defined.

https://github.com/confidential-containers/cloud-api-adaptor/blob/a771f96274c8b98f9f450b9477ad0572902ae94e/pkg/adaptor/cloud/cloud.go#L36-L40

I will raise an PR to refactor the interface after this PR gets merged. Probably, we need the methods below, and initialization code will be moved to `Setup`.
```
 	Setup(ctx context.Context) error 
 	Teardown(ctx context.Context) error 
```